This is from http://bugzilla.xfce.org/show_bug.cgi?id=8064 and will require at least
http://bugzilla.xfce.org/show_bug.cgi?id=8064#c2 too for current xfce4-dev-tools
compability.

Thanks,
Samuli

--- configure.in
+++ configure.in
@@ -30,10 +30,10 @@
 XDT_I18N([ar da el en_GB es eu fr gl hu ja lv nl pt_BR pt_PT tr uk ur])
 
 dnl Check for required packages
-XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.6.0])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.99.2])
-XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.3.99.2])
-XDT_CHECK_PACKAGE([LIBXFCE4PANEL],[libxfce4panel-1.0],[4.3.99.2])
+XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.24.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.10.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.10.0])
+XDT_CHECK_PACKAGE([LIBXFCE4PANEL],[libxfce4panel-1.0],[4.10.0])
 
 #CFLAGS="$CFLAGS -Wall -Werror"
 
--- panel-plugin/datetime.c
+++ panel-plugin/datetime.c
@@ -28,7 +28,7 @@
 #include <string.h>
 
 /* xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libxfce4util/libxfce4util.h>
 #include <libxfce4panel/xfce-panel-plugin.h>
 #include <libxfce4panel/xfce-panel-convenience.h>
--- panel-plugin/datetime-dialog.c
+++ panel-plugin/datetime-dialog.c
@@ -28,7 +28,7 @@
 #include <string.h>
 
 /* xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libxfce4util/libxfce4util.h>
 #include <libxfce4panel/xfce-panel-plugin.h>
 
@@ -388,7 +388,7 @@
   /*
    * layout frame
    */
-  frame = xfce_create_framebox(_("Layout"), &bin);
+  frame = xfce_gtk_frame_box_new(_("Layout"), &bin);
   gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dlg)->vbox), frame,
       FALSE, FALSE, 0);
   gtk_container_set_border_width(GTK_CONTAINER(frame), 6);
@@ -422,7 +422,7 @@
   /*
    * Date frame
    */
-  datetime->date_frame = xfce_create_framebox(_("Date"), &bin);
+  datetime->date_frame = xfce_gtk_frame_box_new(_("Date"), &bin);
   gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dlg)->vbox), datetime->date_frame,
       FALSE, FALSE, 0);
   gtk_container_set_border_width(GTK_CONTAINER(datetime->date_frame), 6);
@@ -525,7 +525,7 @@
   /*
    * time frame
    */
-  datetime->time_frame = xfce_create_framebox(_("Time"), &bin);
+  datetime->time_frame = xfce_gtk_frame_box_new(_("Time"), &bin);
   gtk_box_pack_start(GTK_BOX(GTK_DIALOG(dlg)->vbox), datetime->time_frame,
       FALSE, FALSE, 0);
   gtk_container_set_border_width(GTK_CONTAINER(datetime->time_frame), 6);
--- panel-plugin/Makefile.am
+++ panel-plugin/Makefile.am
@@ -14,7 +14,7 @@
 	-DLOCALEDIR=\"$(localedir)\"		\
 	$(GTK_CFLAGS)				\
 	$(LIBXFCE4PANEL_CFLAGS)			\
-	$(LIBXFCEGUI4_CFLAGS)			\
+	$(LIBXFCE4UI_CFLAGS)			\
 	$(LIBXFCE4UTIL_CFLAGS)
 
 libdatetime_la_LDFLAGS = 			\
@@ -29,7 +29,7 @@
 libdatetime_la_LIBADD = 			\
 	$(GTK_LIBS)				\
 	$(LIBXFCE4PANEL_LIBS)			\
-	$(LIBXFCEGUI4_LIBS)			\
+	$(LIBXFCE4UI_LIBS)			\
 	$(LIBXFCE4UTIL_LIBS)
 
 desktopdir = $(datadir)/xfce4/panel-plugins
